src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,357 @@
+
+# Examples
+```jldoctest
+julia> M = fano_matroid()
+julia> matroid_stratum_matrix_coordinates(M, [1,2,4], GF(2))
+([1 0 x[1, 1] 0 x[1, 2] 0 x[1, 4]; 0 1 x[2, 1] 0 0 x[2, 3] x[2, 4]; 0 0 0 1 x[3, 2] x[3, 3] x[3, 4]], localization of Multivariate Polynomial Ring in 9 variables x[1, 1], x[2, 1], x[1, 2], x[3, 2], ..., x[3, 4] over Galois field with characteristic 2 at the powers of gfp_mpoly[x[3, 3], x[1, 1]*x[2, 3]*x[3, 4] + x[1, 1]*x[3, 3]*x[2, 4] + x[2, 1]*x[3, 3]*x[1, 4], x[2, 3]*x[1, 4], x[1, 2]*x[2, 3]*x[3, 4] + x[1, 2]*x[3, 3]*x[2, 4] + x[3, 2]*x[2, 3]*x[1, 4], x[3, 2], x[1, 1]*x[3, 2]*x[2, 4] + x[2, 1]*x[1, 2]*x[3, 4] + x[2, 1]*x[3, 2]*x[1, 4], x[1, 2]*x[2, 4], x[2, 1], x[1, 1]*x[3, 4]], ideal in localization of Multivariate Polynomial Ring in 9 variables x[1, 1], x[2, 1], x[1, 2], x[3, 2], ..., x[3, 4] over Galois field with characteristic 2 at the powers of gfp_mpoly[x[3, 3], x[1, 1]*x[2, 3]*x[3, 4] + x[1, 1]*x[3, 3]*x[2, 4] + x[2, 1]*x[3, 3]*x[1, 4], x[2, 3]*x[1, 4], x[1, 2]*x[2, 3]*x[3, 4] + x[1, 2]*x[3, 3]*x[2, 4] + x[3, 2]*x[2, 3]*x[1, 4], x[3, 2], x[1, 1]*x[3, 2]*x[2, 4] + x[2, 1]*x[1, 2]*x[3, 4] + x[2, 1]*x[3, 2]*x[1, 4], x[1, 2]*x[2, 4], x[2, 1], x[1, 1]*x[3, 4]] generated by the elements 0//1, (x[2, 3]*x[3, 4] + x[3, 3]*x[2, 4])//1, (x[1, 2]*x[3, 4] + x[3, 2]*x[1, 4])//1, (x[1, 1]*x[2, 4] + x[2, 1]*x[1, 4])//1, (x[1, 1]*x[3, 2]*x[2, 3] + x[2, 1]*x[1, 2]*x[3, 3])//1)
+
+```
+"""
+
+
+
+function matroid_stratum_matrix_coordinates(M::Matroid, B::Vector{Int64},
+                                            F::AbstractAlgebra.Ring = ZZ)
+    d = rank(M)
+    n = length(matroid_groundset(M))
+    Bs = bases(M)
+    R, x, xdict = make_polynomial_ring(Bs,B,F)
+    return matroid_stratum_matrix_coordinates_given_ring(d, n, M, F, B, R, x, xdict)
+end
+
+    
+function realization_bases_coordinates(Bs::Vector{Vector{Int64}}, A::Vector{Int64})
+
+    d = length(Bs[1])
+    B = A[1:d]
+    c1 = A[d+1]
+
+
+    coordBases = [b for b in Bs if length(symdiff(B,b)) == 2]
+    
+    newCoords = Vector{Vector{Int64}}([])
+    
+    for b in coordBases
+        if is_subset(b, A)
+            continue
+        end
+        
+        row_b = setdiff(B,b)[1]
+        row_b = length([a for a in B if a < row_b]) + 1
+        
+        col_b = setdiff(b,B)[1]
+        col_b = col_b - length([a for a in A if a <= col_b])
+
+        push!(newCoords, [row_b, col_b])
+    end
+    
+    return sort!(newCoords, by = x -> (x[2], x[1]))
+end
+
+
+function partial_matrix_max_rows(Vs::Vector{Vector{Int64}})
+    nr = maximum([x[1] for x in Vs])
+    cols = unique!([x[2] for x in Vs])
+    first_nonzero_cols = Dict{Int64, Int64}([c => maximum([i for i in 1:nr if [i,c] in Vs]) for c in cols])
```

Comment:
Again you can avoid allocations (and gain speed) by switching from using arrays/vectors to using generator expressions:
```suggestion
    first_nonzero_cols = Dict{Int64, Int64}(c => maximum(i for i in 1:nr if [i,c] in Vs) for c in cols)
```
Not that it matters much here, but I think it's good to learn about these things hence I mention it :-)

Reply:
Again this applies to multiple other places, too